protocol/contracts/libraries/Silo/LibLegacyTokenSilo.sol

```diff
@@ -0,0 +1,600 @@
+ 
+                // add to new silo
+                LibTokenSilo.addDepositToAccount(
+                    account, 
+                    perTokenData.token, 
+                    LibTokenSilo.grownStalkAndBdvToStem(
+                        IERC20(perTokenData.token), 
+                        perDepositData.grownStalk,
+                        crateBDV
+                    ), 
+                    perDepositData.amount, 
+                    crateBDV,
+                    LibTokenSilo.Transfer.emitTransferSingle
+                );
+ 
+                // add to running total of seeds
+                migrateData.totalSeeds = migrateData.totalSeeds.add(crateBDV.mul(getSeedsPerToken(address(perTokenData.token))).toUint128());
+
+                // emit legacy RemoveDeposit event
+                emit RemoveDeposit(account, perTokenData.token, perDepositData.season, perDepositData.amount);
+            }
+ 
+            // init mow status for this token
+            setMowStatus(account, perTokenData.token, perTokenData.stemTip);
+        }
+ 
+        // user deserves stalk grown between stemStartSeason and now
+        LibSilo.mintGrownStalk(account, migrateData.totalGrownStalk);
+
+        //return seeds diff for checking in the "part 2" of this function (stack depth kept it from all fitting in one)
+        return balanceOfSeeds(account).sub(migrateData.totalSeeds);
+    }
+
+    function _mowAndMigrateMerkleCheck(
+        address account,
+        uint256 stalkDiff,
+        uint256 seedsDiff,
+        bytes32[] calldata proof,
+        uint256 seedsVariance
+    ) internal {
+        if (seedsDiff > 0) {
+            //verify merkle tree to determine stalk/seeds diff drift from convert issue
+            bytes32 leaf = keccak256(abi.encode(account, stalkDiff, seedsDiff));
+            
+            require(
+                MerkleProof.verify(proof, DISCREPANCY_MERKLE_ROOT, leaf),
+                "UnripeClaim: invalid proof"
+            );
+        }
+
+        //make sure seedsVariance equals seedsDiff input
+        require(seedsVariance == seedsDiff, "seeds misalignment, double check submitted deposits");
+
+        AppStorage storage s = LibAppStorage.diamondStorage();
+
+
+        //and wipe out old seed balances (all your seeds are belong to stem)
+        setBalanceOfSeeds(account, 0);
+
+        //emit that all their seeds are gone
+        emit SeedsBalanceChanged(account, -int256(s.a[account].s.seeds));
+
+        //stalk drive was calculated based on ENROOT_FIX_SEASON, so we need to calculate
+        //the amount of stalk that has grown since then
+        uint256 currentStalkDiff = (uint256(s.season.current).sub(ENROOT_FIX_SEASON)).mul(seedsDiff).add(stalkDiff);
+
+        //emit the stalk variance
+        if (currentStalkDiff > 0) {
+            emit StalkBalanceChanged(account, -currentStalkDiff.toInt256(), 0);
```

Comment:
Does it need to decrement the Stalk in addition to emitting an event?

Reply:
Yes, fixed here: https://github.com/BeanstalkFarms/Beanstalk/commit/4fb20bc7bad96a29570a6e14c352a86c1ba0e478